cmake-tool/helpers/cpio.cmake

```diff
@@ -36,11 +36,11 @@ endfunction()
 # into another target
 function(MakeCPIO output_name input_files)
     cmake_parse_arguments(PARSE_ARGV 2 MAKE_CPIO "" "CPIO_SYMBOL" "DEPENDS")
-    if(NOT "${MAKE_CPIO_UNPARSED_ARGUMENTS}" STREQUAL "")
```

Comment:
This pattern is repeated a lot throughout the CMake files.  I recall it's partially because `if (${MAKE_CPIO_UNPARSED_ARGUMENTS})` unexpectedly evaluates to false for almost all string values as  CMake's described behavior (https://cmake.org/cmake/help/latest/command/if.html).

Using `if(MAKE_CPIO_UNPARSED_ARGUMENTS)` is shorter and you know that this tests if the variable is defined and defined to non-false values. But it would be easy to misread something like `if(${MAKE_CPIO_UNPARSED_ARGUMENTS}` or `if("${MAKE_CPIO_UNPARSED_ARGUMENTS}"`) as testing for truth  if you aren't more familiar with the if() gotchas . In these cases, if the variable contains a string it will always evaluate to false unless the contents match the truthy string values (eg ON, TRUE, etc). So while `"${MAKE_CPIO_UNPARSED_ARGUMENTS}" STREQUAL "")` is more verbose, it's probably a good pattern to leave around for explicitly testing for a non-empty string as it doesn't rely on understanding any if() testing gotchas.

Reply:
I agree with all points. This is the outcame from digging into CMake usage and a long term plan to use the language as it seems intended nowadays. I can move this to a separate PR to make the intention more clear. There are more places where this could be changed to clarify what checks are non-trivial.

Reply:
I have remove the commit